weasyl/controllers/profile.py

```diff
@@ -1,526 +1,518 @@
+    rating = define.get_rating(request.userid)
+    otherid = profile.resolve(request.userid, form.userid, form.name)
+    folderid = define.get_int(form.folderid) or None
+
+    if not otherid:
+        raise WeasylError("userRecordMissing")
+    elif not request.userid and "h" in define.get_config(otherid):
+        return Response(define.errorpage(request.userid, errorcode.no_guest_access))
+
+    userprofile = profile.select_profile(otherid, images=True, viewer=request.userid)
+    has_fullname = userprofile['full_name'] is not None and userprofile['full_name'].strip() != ''
+    page_title = u"%s's submissions" % (userprofile['full_name'] if has_fullname else userprofile['username'],)
+    page = define.common_page_start(request.userid, title=page_title)
+
+    url_format = "/submissions/{username}?%s{folderquery}".format(
+                 username=define.get_sysname(userprofile['username']),
+                 folderquery="&folderid=%d" % folderid if folderid else "")
+    result = pagination.PaginatedResult(
+        submission.select_list, submission.select_count, 'submitid', url_format, request.userid, rating,
+        60, otherid=otherid, folderid=folderid, backid=define.get_int(form.backid),
+        nextid=define.get_int(form.nextid), config=config, profile_page_filter=not folderid)
+
+    page.append(define.render('user/submissions.html', [
+        # Profile information
+        userprofile,
+        # User information
+        profile.select_userinfo(otherid, config=userprofile['config']),
+        # Relationship
+        profile.select_relation(request.userid, otherid),
+        # Recent submissions
+        result,
+        # Folders
+        folder.select_list(otherid, "sidebar/all"),
+        # Current folder
+        folderid,
+    ]))
+
+    return Response(define.common_page_end(request.userid, page))
+
+
+def collections_(request):
+    form = request.web_input(userid="", name="", backid=None, nextid=None,
+                     folderid=None)
+    form.name = request.matchdict.get('name', form.name)
+    form.userid = define.get_int(form.userid)
+
+    config = define.get_config(request.userid)
+    rating = define.get_rating(request.userid)
+    otherid = profile.resolve(request.userid, form.userid, form.name)
+
+    if not otherid:
+        raise WeasylError("userRecordMissing")
+    elif not request.userid and "h" in define.get_config(otherid):
+        return Response(define.errorpage(request.userid, errorcode.no_guest_access))
+
+    userprofile = profile.controller_baseselect_profile(otherid, images=True, viewer=request.userid)
+    has_fullname = userprofile['full_name'] is not None and userprofile['full_name'].strip() != ''
+    page_title = u"%s's collections" % (userprofile['full_name'] if has_fullname else userprofile['username'],)
+    page = define.common_page_start(request.userid, title=page_title)
+
+    url_format = "/c if name else ollections?userid={userid}&%s".format(userid=userprofile['userid'])
```

Comment:
Thanks for catching that. That's a bad trackpad driver doing a lot of pasting when I scroll in pycharm.